Nexus SSH: Support another magic IP address line format

mirceaulinic opened the pull request:
Turns out that besides the following format:
``IPv6 link-local address: fe80::2ec2:60ff:fe4f:feb2 (default) [VALID]``,
Cisco Nexus also presents the IPv6 addresses under the following format:
``IPv6 address: fe80::2ec2:60ff:fe4f:feb2 [VALID]``.

ktbyers commented on napalm/nxos_ssh/nxos_ssh.py:
```diff
@@ -903,14 +903,22 @@ def get_interfaces_ip(self):
+                    # match the following format:
+                    # IPv6 address: fe80::a293:51ff:fe5f:5ce9 [VALID]
+                    ip_address = line.split()[2]
+                    prefix_len = "64"
```
Are addresses of this form always link-local? Should we put a check in where if `ip_address` contains a `/` then raise an exception. I am a bit worried that we might have a case where we incorrectly assume a /64 prefix when it isn't.

Maybe a better solution would just be to move the case where we search for '/' up (move it up from the else clause and change the logic a bit). Structurally something like:

 ```
if "link-local address" in line:
    ip_address = line.split()[3]
    prefix_len = "64"
elif "/" in line:
    ip_address, prefix_len = line.split()[0].split("/")
elif "IPv6 address" in line:
    ip_address = line.split()[2]
    prefix_len = "64"
else:
    raise ValueError("We don't know what just happened here")
```

mirceaulinic commented:
Yes, from what I've seen across a dozen of devices, only link-local have
that format.

On Wed, Jan 16, 2019, 18:00 Kirk Byers <notifications@github.com> wrote:

> *@ktbyers* commented on this pull request.
> ------------------------------
>
> In napalm/nxos_ssh/nxos_ssh.py
> <https://github.com/napalm-automation/napalm/pull/912#discussion_r248389674>
> :
>
> >                      ip_address = line.split()[3]
>                      prefix_len = "64"
> +                elif "IPv6 address" in line:
> +                    # match the following format:
> +                    # IPv6 address: fe80::a293:51ff:fe5f:5ce9 [VALID]
> +                    ip_address = line.split()[2]
> +                    prefix_len = "64"
>
> Are addresses of this form always link-local? Should we put a check in
> where if ip_address contains a / then raise an exception. I am a bit
> worried that we might have a case where we incorrectly assume a /64 prefix
> when it isn't.
>
> Maybe a better solution would just be to move the case where we search for
> '/' up (move it up from the else clause and change the logic a bit).
> Structurally something like:
>
> if "link-local address" in line:
>    ip_address = line.split()[3]
>    prefix_len = "64"
> elif "/" in line:
>    ip_address, prefix_len = line.split()[0].split("/")
> elif "IPv6 address" in line:
>    ip_address = line.split()[2]
>    prefix_len = "64"
> else:
>    raise ValueError("We don't know what just happened here")
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/napalm-automation/napalm/pull/912#pullrequestreview-193273633>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AP69lyGx8RJH9lH8tuI0zrLj3xwblIz8ks5vD2ijgaJpZM4aCwd9>
> .
>
